src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts

```diff
@@ -90,16 +107,28 @@ export class TableDesignerComponentInput implements DesignerComponentInput {
 			message: localize('tableDesigner.generatingScript', "Generating script..."),
 			sticky: true
 		});
+		const telemetryInfo = {
+			provider: this._provider.providerId,
+			isNewTable: this._tableInfo.isNewTable,
+		};
+		Object.assign(telemetryInfo, this._telemetryInfo);
+		const saveEvent = this._adsTelemetryService.createActionEvent(TelemetryView.TableDesigner, TelemetryAction.GenerateScript).withAdditionalProperties(telemetryInfo);
+		const startTime = new Date().getTime();
 		try {
 			this.updateState(this.valid, this.dirty, 'generateScript');
 			const script = await this._provider.generateScript(this._tableInfo);
 			this._queryEditorService.newSqlEditor({ initalContent: script });
 			this.updateState(this.valid, this.dirty);
 			notificationHandle.updateMessage(localize('tableDesigner.generatingScriptCompleted', "Script generated."));
+			saveEvent.withAdditionalMeasurements({
+				'elapsedTimeMs': new Date().getTime() - startTime
+			}).send();
 		} catch (error) {
 			notificationHandle.updateSeverity(Severity.Error);
 			notificationHandle.updateMessage(localize('tableDesigner.generateScriptError', "An error occured while generating the script: {0}", error?.message ?? error));
 			this.updateState(this.valid, this.dirty);
+			this._adsTelemetryService.createErrorEvent(TelemetryView.TableDesigner, TelemetryAction.GenerateScript,
+				error?.code, error?.message).withAdditionalProperties(telemetryInfo).send();
```

Comment:
please do not include message here.

Reply:
![Screen Shot 2022-02-01 at 8 04 10 PM](https://user-images.githubusercontent.com/6411451/152091568-ab8afd5f-eb3e-4c14-add2-ebdff105c95e.png)

I verified that the message doesn't contain PII. The stack trace does.

Reply:
this is just an exception that happens not contain anything sensitive in its message, we can't guarantee other types of exceptions, Yesterday, you said you are going to check whether the telemetry service will remove sensitive information automatically, did you find anything? if not, let's remove the message.

Reply:
ping...